samples/java-spring-eventsourced-customer-registry-subscriber/src/it/java/customer/api/CustomerIntegrationTest.java

```diff
@@ -1,92 +1,171 @@
+ */
 @SpringBootTest(classes = Main.class)
-@Import(TestkitConfig.class)
-public class CustomerIntegrationTest extends KalixIntegrationTestKitSupport {
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CustomerIntegrationTest {
+
+  final private Duration timeout = Duration.of(5, SECONDS);
+
+  private DockerComposeUtils dockerComposeUtils = new DockerComposeUtils("docker-compose-integration.yml");
+
+  private KalixSpringApplication kalixSpringApplication;
+
+  public CustomerIntegrationTest(ApplicationContext applicationContext) {
+    Map<String, Object> confMap = new HashMap<>();
+    confMap.put("kalix.user-function-port", dockerComposeUtils.userFunctionPort());
+    // don't kill the test JVM when terminating the KalixRunner
+    confMap.put("kalix.system.akka.coordinated-shutdown.exit-jvm", "off");
+    // dev-mode should be false when running integration tests
+    confMap.put("kalix.dev-mode.enabled", false);
+    confMap.put("kalix.user-function-interface", "0.0.0.0");
+
+    // read service-port-mappings and pass to UF
+    dockerComposeUtils.getLocalServicePortMappings().forEach(entry -> {
+        var split = entry.replace("-D", "").split("=");
+        confMap.put(split[0], split[1]);
+      }
+    );
 
+    Config config = ConfigFactory.parseMap(confMap).withFallback(ConfigFactory.load());
 
+    kalixSpringApplication = new KalixSpringApplication(applicationContext, config);
+  }
 
-  @Autowired
-  private WebClient webClient;
+  @BeforeAll
+  public void beforeAll() {
+    dockerComposeUtils.start();
+    kalixSpringApplication.start();
+  }
 
-  private Duration timeout = Duration.of(5, SECONDS);
+  @AfterAll
+  public void afterAll() {
+    kalixSpringApplication.stop();
+    dockerComposeUtils.stop();
+  }
 
-  private void assertSourceServiceIsUp(WebClient sourceServiceWebClient) {
+  private HttpStatusCode assertSourceServiceIsUp(WebClient webClient) {
     try {
-      var code =
-          sourceServiceWebClient.get()
-              .retrieve()
-              .onStatus(HttpStatusCode::is4xxClientError, clientResponse ->
-                  Mono.empty()
-              )
-              .toBodilessEntity()
-              .block(timeout);;
-      Assertions.assertEquals(HttpStatus.NOT_FOUND, code.getStatusCode()); // NOT_FOUND is a sign that the source service is there
+      return webClient.get()
+        .retrieve()
+        .onStatus(HttpStatusCode::is4xxClientError, clientResponse ->
+          Mono.empty()
+        )
+        .toBodilessEntity()
+        .block(timeout)
+        .getStatusCode();
+
     } catch (WebClientRequestException ex) {
-      Assertions.fail("This test requires an external kalix service to be running on localhost:9000 but was not able to reach it.");
+      throw new RuntimeException("This test requires an external kalix service to be running on localhost:9001 but was not able to reach it.");
```

Comment:
The customer registry still runs on 9000, no?

```suggestion
      throw new RuntimeException("This test requires an external kalix service to be running on localhost:9000 but was not able to reach it.");
```